### DISCUSSION TITLE
Null symbol in style tag

### DISCUSSION THREAD

**Discussion by IKolosynskyi** [emotion-js/emotion#2632] [Q&A]
Hi!

I am using @emotion,

```
    "next": "^9.5.5",
    "@emotion/cache": "^11.7.1",
    "@emotion/react": "^11.7.1",
    "@emotion/styled": "^11.6.0",
```

with CacheProvider
```
export const emotionCache = createCache({
	key: "css-cache",
	speedy: false,
});
```

Also, the web site is connected to yandex webvisor which, in some way, crawl and grab style tags.
That's how I see style tag
![image](https://user-images.githubusercontent.com/11165501/152546394-510cbf85-c6f2-4b95-ae2e-642474be8227.png)
That's how it looks like in webvisor
![image](https://user-images.githubusercontent.com/11165501/152545916-163c7d51-1b75-40a1-bec1-e7a9d4f05806.png)

Have you ever seen these symbols in style tag? If no, is there any way to provide own render function in which I can define the style tag format so I will be sure that there is no other artifacts except css and data attribute. This data-s looks strange to me.

Thanks in advance!

**Comment by srmagura** [comment]
[Here](https://github.com/emotion-js/emotion/blob/main/packages/sheet/src/index.js#L61) is the code that adds the `data-s` attribute. As you can see, the attribute's value is an empty string — there are no weird characters.

This seems like a Yandex problem to me, but let me know if you think otherwise.

**Reply by IKolosynskyi** [reply]
Thanks, I will analyze this question a bit more but looks like it's really not related to the library. 
But I am still curios, why do you need  empty data-s attribute?

**Reply by srmagura** [reply]
My understanding from reading the code is that it is an internal marker used by Emotion to facilitate hydration of server-rendered `<style>` tags.

**Reply by Andarist** [reply]
Yes, exactly - this is a marker added during hydration that makes sure that no other instance of Emotion tries to hydrate those already hydrated elements.

**Reply by Andarist** [reply]
That being said - I'm not sure how it's reported like this. We are setting the value of this attribute to an empty string:
https://github.com/emotion-js/emotion/blob/d25316393639232df16ba836b407e3678eea5e4d/packages/cache/src/index.js#L80
